Xwt/Xwt/Toolkit.cs

```diff
@@ -375,6 +375,10 @@ public object ValidateObject (object obj)
 		{
 			if (obj is Image)
 				((Image)obj).InitForToolkit (this);
+			else if (obj is TextLayout)
+				((TextLayout)obj).InitForToolkit (this);
+			else if (obj is Font)
+				return ((IFrontend)obj).ToolkitEngine != this ? Font.FromName (obj.ToString (), this) : obj;
```

Comment:
This is not very efficient. If a Font is created in for the GTK toolkit and then used intensively in Mac it will result in fonts being constantly recreated.